### PR TITLE
Fix getting time zone names with Invariant Culture

### LINKED ISSUE #33037: TimeZoneInfo StandardName, DisplayName changes
Steps to reproduce the issue
use Net Core 2.1 and Display StandardName and DisplayName on Windows os
use Net Core 3.1 and Display StandardName and DisplayName on Linux os

Expected behavior
We expect that the StandardNamas and DisplayNames to be the same from 2.1 to 3.1

Actual behavior
It seams that the behavior has changed from 2.1 to 3.1 on Linux os, the DisplayName is replaced by StandardName and vice versa.

### PR DESCRIPTION
**Pull request by tarekgh**
Fixes #33037

### CODE REVIEW

**Comment by tarekgh** [comment]
@eerhardt could you please have a look? Thanks.

CC @danmosemsft @cristina-dorotan

File: src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs [hunk 1]
```diff
@@ -2268,6 +2268,29 @@ public static void EnsureUtcObjectSingleton()
...
 
+        [Fact]
+        public static void TestNameWithInvariantCulture()
+        {
+            CultureInfo ci = CultureInfo.CurrentUICulture;
+            try
+            {
+                // We call ICU to get the names. When passing invariant culture name to ICU, it fail and we'll use the abbreviated names at that time.
+                // We fixed this issue by avoid sending the invariant culture name to ICU and this test is confirming we work fine at that time.
+                CultureInfo.CurrentUICulture = CultureInfo.InvariantCulture;
```
Comment (eerhardt) [inline src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs]: Should you use RemoteExecutor so this doesn't affect other test cases?

File: src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs [hunk 2]
```diff
@@ -80,9 +80,10 @@ private TimeZoneInfo(byte[] data, string id, bool dstDisabled)
             }
             _displayName = _standardDisplayName;
 
-            GetDisplayName(Interop.Globalization.TimeZoneDisplayNameType.Generic, ref _displayName);
-            GetDisplayName(Interop.Globalization.TimeZoneDisplayNameType.Standard, ref _standardDisplayName);
-            GetDisplayName(Interop.Globalization.TimeZoneDisplayNameType.DaylightSavings, ref _daylightDisplayName);
+            string uiCulture = CultureInfo.CurrentUICulture.Name.Length == 0 ? "en-US" : CultureInfo.CurrentUICulture.Name; // ICU doesn't work nicely with Invariant
```
Comment (eerhardt) [inline src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs]: Maybe make `en-US` a `private const` and call it `FallbackCultureName` or similar?

File: src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs [hunk 3]
```diff
@@ -125,11 +126,28 @@ private unsafe void GetDisplayName(Interop.Globalization.TimeZoneDisplayNameType
                         return Interop.Globalization.GetTimeZoneDisplayName(locale, id, type, bufferPtr, buffer.Length);
                     }
                 },
-                CultureInfo.CurrentUICulture.Name,
+                uiCulture,
                 _id,
                 nameType,
                 out timeZoneDisplayName);
 
+            if (!result && CultureInfo.CurrentUICulture.Name != "en-US")
```
Comment (eerhardt) [inline src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs]:
```suggestion
            if (!result && uiCulture != "en-US")
```

File: src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs [hunk 4]
```diff
@@ -2268,6 +2269,30 @@ public static void EnsureUtcObjectSingleton()
             Assert.True(ReferenceEquals(TimeZoneInfo.FindSystemTimeZoneById("UTC"), TimeZoneInfo.Utc));
         }
 
+        // We test the exsitance of a specific English time zone name to avoid failures on non-English platforms.
```
Comment (Wraith2) [inline src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs]: exsitance -> existance

File: src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs [hunk 5]
```diff
@@ -2268,6 +2269,30 @@ public static void EnsureUtcObjectSingleton()
...
+            {
+                CultureInfo ci = CultureInfo.CurrentUICulture;
+                // We call ICU to get the names. When passing invariant culture name to ICU, it fail and we'll use the abbreviated names at that time.
+                // We fixed this issue by avoid sending the invariant culture name to ICU and this test is confirming we work fine at that time.
+                CultureInfo.CurrentUICulture = CultureInfo.InvariantCulture;
+                TimeZoneInfo.ClearCachedData();
+
+                TimeZoneInfo pacific = TimeZoneInfo.FindSystemTimeZoneById(s_strPacific);
+
+                Assert.True(pacific.StandardName.IndexOf("Pacific", StringComparison.OrdinalIgnoreCase) >= 0, $"'{pacific.StandardName}' is not the expeted standard name for Pacific time zone");
```
Comment (Wraith2) [inline src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs]: expeted -> expected

**Comment by tarekgh** [comment]
@eerhardt please let me know if you have any more feedback. Thanks for your review.

**Review by Wraith2 — APPROVED** [review]

File: src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs [hunk 6]
```diff
@@ -2268,6 +2269,30 @@ public static void EnsureUtcObjectSingleton()
             Assert.True(ReferenceEquals(TimeZoneInfo.FindSystemTimeZoneById("UTC"), TimeZoneInfo.Utc));
         }
 
+        // We test the existance of a specific English time zone name to avoid failures on non-English platforms.
+        [ConditionalFact(nameof(IsEnglishUILanguage))]
+        public static void TestNameWithInvariantCulture()
+        {
+            RemoteExecutor.Invoke(() =>
+            {
+                CultureInfo ci = CultureInfo.CurrentUICulture;
```
Comment (eerhardt) [inline src/libraries/System.Runtime/tests/System/TimeZoneInfoTests.cs]: (nit) this doesn't appear to be used anymore.

**Review by eerhardt — APPROVED** [review]
Looks good to me. Thanks for fixing this, @tarekgh.

**Pull request merged**